### PR TITLE
Bug 1150009 - Add cleanup functions to the gaia UI tests that install we...

### PR DESCRIPTION
**Pull request by mwargers**
...bapps

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=31f0353086e77cd00e64f07ebdec63e8cc0bb900

**Comment by davehunt** [comment]
This moves the cleanup of additional apps to the tear down, whereas the rest of our cleanup is still in set up. I would prefer to explicitly uninstall any apps in the individual tests tear down, but I understand that if there's a failure in the set up or earlier in the tear down then the apps will not be uninstalled.

Could we change the cleanup to filter the apps deleted to exclude the bundled apps? I suspect there'd be a way to identify these, based on https://developer.mozilla.org/en-US/docs/DOMApplication

**Comment by mwargers** [comment]
I wouldn't know how to uninstall the app on teardown without triggering the uninstall dialog when using navigator.mozApps.uninstall.
Supposedly by setting the dom.mozApps.auto_confirm_uninstall pref, I guess.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4e7b0e6167efc97eccd3e4987310622cc7a45b9a

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4e7b0e6167efc97eccd3e4987310622cc7a45b9a

File: tests/python/gaia-ui-tests/gaiatest/tests/functional/homescreen/test_homescreen_delete_app.py [hunk 1]
```diff
@@ -51,3 +51,13 @@ def test_delete_app(self):
         # Check that the app is no longer available
         with self.assertRaises(AssertionError):
             self.apps.launch(self.test_data['name'])
+
+    def tearDown(self):
+        self.marionette.switch_to_frame()
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/tests/functional/homescreen/test_homescreen_delete_app.py]: There's a lot of duplication here. Could we move this to a `GaiaApps.uninstall` method?

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8326168442e36d83347c2cf5c244316e75cbb187

File: tests/python/gaia-ui-tests/gaiatest/gaia_test.py [hunk 2]
```diff
@@ -106,6 +106,15 @@ def kill_all(self):
         self.marionette.switch_to_frame()
         self.marionette.execute_async_script("GaiaApps.killAll();")
 
+    def uninstall(self, name):
+        self.marionette.switch_to_frame()
+        data_layer = GaiaData(self.marionette)
+        data_layer.set_bool_pref('dom.mozApps.auto_confirm_uninstall', True)
+        result = self.marionette.execute_async_script('GaiaApps.uninstallWithName("%s")' % name)
+        if not result is True:
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/gaia_test.py]: Why don't we use `assert` like done in [kill](https://github.com/mozilla-b2g/gaia/pull/29550/files#diff-6d544bbb7c759ffb68f344f348f7740eL96) just above?

Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/gaia_test.py]: Sure, I can do that.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3c83bcdb3944c980230cac0fd7b1a35e41dd64c1

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b56324922a8ce4571e688245fd7d7293c983a549

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5dcc3c9a346f1b34aef64df86259f32c21269de2

**Pull request merged**